lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -547,46 +550,103 @@ DiagnosticVerifier::Result DiagnosticVerifier::verifyFile(unsigned BufferID) {
     
     auto &FoundDiagnostic = *FoundDiagnosticIter;
 
-    const char *IncorrectFixit = nullptr;
+    const char *missedFixitLoc = nullptr;
     // Verify that any expected fix-its are present in the diagnostic.
     for (auto fixit : expected.Fixits) {
       // If we found it, we're ok.
-      if (!checkForFixIt(fixit, FoundDiagnostic, InputFile))
-        IncorrectFixit = fixit.StartLoc;
+      if (!checkForFixIt(fixit, FoundDiagnostic, InputFile)) {
+        missedFixitLoc = fixit.StartLoc;
+        break;
+      }
     }
 
-    bool matchedAllFixIts =
-        expected.Fixits.size() == FoundDiagnostic.FixIts.size();
+    bool isUnexpectedFixitsSeen =
+        expected.Fixits.size() < FoundDiagnostic.FixIts.size();
+
+    auto makeActualFixitsPhrase =
+        [&](ArrayRef<DiagnosticInfo::FixIt> actualFixits,
+            std::string &actualFixitsStr) -> std::string {
+      actualFixitsStr = renderFixits(actualFixits, InputFile);
+
+      std::string str("actual fix-it");
+      if (actualFixits.size() >= 2) {
+        str += "s";
+      }
+      str += " seen: ";
+      str += actualFixitsStr;
```

Comment:
This is fine, but using `llvm::Twine` might make the concatenation a little more efficient

Reply:
Thanks! I did it.